### PR TITLE
ol 2.4 (new formula)

### PR DESCRIPTION
**Pull request by yuriy-chumak**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/ol.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class Ol < Formula
+  desc "Purely functional dialect of Lisp"
+  homepage "https://yuriy-chumak.github.io/ol/"
+  url "https://github.com/yuriy-chumak/ol/archive/2.4.tar.gz"
```
[the remark this answers is not in the document]
Comment (yuriy-chumak) [reply Formula/ol.rb:4]: Done.

File: Formula/ol.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  sha256 "019978ddcf0befc8b8de9f50899c9dd0f47a3e18cf9556bc72a75ae2d1d965d4"
+  license any_of: ["LGPL-3.0-or-later", "MIT"]
+
```
[the remark this answers is not in the document]
Comment (yuriy-chumak) [reply Formula/ol.rb:7]: Done.

File: Formula/ol.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  uses_from_macos "vim" => :build # for xxd
+
+  def install
+    system "make", "all", "PREFIX=#{prefix}"
+    system "make", "install", "PREFIX=#{prefix}"
+  end
+
+  test do
+    system "#{bin}/ol", "--version"
```
[the remark this answers is not in the document]
Comment (yuriy-chumak) [reply Formula/ol.rb:16]: Added `assert_equal` with *gcd(1071, 1029)* calculation.

File: Formula/ol.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+  uses_from_macos "vim" => :build # for xxd
+
+  def install
+    system "make", "all", "PREFIX=#{prefix}"
+    system "make", "install", "PREFIX=#{prefix}"
+  end
+
+  test do
+    assert_equal "21", shell_output("echo '(print (gcd 1071 1029))' | #{bin}/ol").strip
```
Comment (ZhongRuoyu) [inline Formula/ol.rb:17]:
```suggestion
    (testpath/"gcd.ol").write <<~EOS
      (print (gcd 1071 1029))
    EOS
    assert_equal "21", shell_output("#{bin}/ol gcd.ol").strip
```

Comment (yuriy-chumak) [reply Formula/ol.rb]: Changed.

**Review by chenrui333 — APPROVED** [review]

**Review by ZhongRuoyu — APPROVED** [review]
Thanks for your first contribution to Homebrew, @yuriy-chumak!

**Pull request closed without merging**